test/dynamo/test_activation_checkpointing.py

```diff
@@ -331,6 +331,117 @@ def fn(x, y, z):
         body_function = getattr(cnt.graphs[0], wrap_node.args[0].name)
         self.assertEqual(op_count(body_function), 2)
 
+    @requires_cuda()
+    def test_ddp(self):
+        import os
+
+        import torch.distributed as dist
+        from torch.nn.parallel import DistributedDataParallel as DDP
+        from torch.testing._internal.common_utils import find_free_port
+
+        N = 1000
+
+        class InnerModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear1 = torch.nn.Linear(N, N)
+                self.linear2 = torch.nn.Linear(N, N)
+
+            def forward(self, x):
+                a = self.linear1(x)
+                a = self.linear2(a)
+                return a
+
+        class MockModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.inner_mod1 = InnerModule()
+                self.inner_mod2 = InnerModule()
+
+            def forward(self, x):
+                a = checkpoint(self.inner_mod1, x, use_reentrant=False)
+                a = torch.cos(a)
+                a = checkpoint(self.inner_mod2, a, use_reentrant=False)
+                a = torch.cos(a)
+                return a
+
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = str(find_free_port())
+        dist.init_process_group("gloo", rank=0, world_size=1)
+
+        mod = MockModule()
+        mod = DDP(mod, bucket_cap_mb=1)
+        x = torch.randn(N, N, requires_grad=True)
+        args = (x,)
+
+        backend = "aot_eager"
+        cnt = CompileCounterWithBackend(backend)
+
+        expected = mod(*args)
+        result = torch.compile(mod, backend=cnt)(*args)
+        dist.destroy_process_group()
```

Comment:
i might prefer you move these tests into test/dynamo/distributed.py and use the infra there that manages processgroup creation/destroy.   You could add a case under `class TestSingleProc(DynamoDistributedSingleProcTestCase):`

Reply:
Yeah, thats probably better. Will move tests there.